lib/bolt/inventory/group.rb

```diff
@@ -18,12 +18,18 @@ class Group
       GROUP_KEYS = DATA_KEYS + %w[name groups targets]
       CONFIG_KEYS = Bolt::Config::INVENTORY_OPTIONS.keys
 
-      def initialize(input, plugins)
+      def initialize(input, plugins, all_group: false)
         @logger = Bolt::Logger.logger(self)
         @plugins = plugins
 
         input = @plugins.resolve_top_level_references(input) if @plugins.reference?(input)
 
+        if all_group && input.key?('name')
+          raise ValidationError.new("Top-level group 'all' cannot specify name '#{input['name']}'", nil)
```

Comment:
Can we warn that the name will be overridden and have this error in 3.0? Just because the current behavior allows it and while it seems like a bug, it'd be a pain for people to rewrite around.